pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -429,6 +431,25 @@ public AtomicLong getPendingWriteOps() {
         return pendingWriteOps;
     }
 
+    public CompletableFuture<Void> unloadSubscription(String subName) {
```

Comment:
Have you considered calling it concurrently, do we have any problem when using multi-thread?

Reply:
- `create new sub` & `unload sub`:  `create a new sub` is executed only when there is no sub, and `unload sub` is performed only when there is a sub. No concurrent calls can occur
- `unload sub` & `unload sub`: I added a concurrent Map to prevent reentrant
- `unload sub` & `close topic`: use `PersistentTopic.lock` to make sure it's correct

I missed the `2&3` two scenes earlier. Thanks for reminding me